src/node.h

```diff
@@ -273,6 +279,13 @@ class NODE_EXTERN MultiIsolatePlatform : public v8::Platform {
   // This function may only be called once per `Isolate`.
   virtual void RegisterIsolate(v8::Isolate* isolate,
                                struct uv_loop_s* loop) = 0;
+  // If your module/application will handle task scheduling on its own through
+  // IsolatePlatformDelegate then you can register with this version. If your
+  // isolate is registered with this overload then you *may not* invoke any
+  // other method in this class against that isolate, except for
+  // `UnregisterIsolate`.
+  virtual void RegisterIsolate(v8::Isolate* isolate,
+                               IsolatePlatformDelegate* delegate);
```

Comment:
Just going to point out for everyone else that this is an ABI-breaking change. However, this API is focused pretty much exclusively on embedders, so it may be okay to make an exception and land this as semver-minor.

Reply:
If we are breaking the ABI anyway and since the usual `RegisterIsolate` is pure virtual..why isn't this pure virtual as well but instead `UNREACHABLE()`? I suppose it's less error-prone to require whoever implements this to make it `UNREACHABLE()` themselves if they want to?

Reply:
The reasoning is that the current patch at least is source compatible. It's true that probably few/zero users have actually inherited from this class so I'm ok with making it pure virtual if you are.

Reply:
I am OK with not making it pure virtual in this patch if we want to keep it semver-minor - making it pure virtual would bit a bit more correct in my opinion, but I think that would tilt the PR a bit more towards semver-major.

Reply:
Suggestion: make it pure virtual and then change it to UNREACHABLE() in the back-port to v13.x?